evalml/automl/automl_search.py

```diff
@@ -495,96 +517,111 @@ def search(self, data_checks="auto", show_iteration_plot=True):
-        logger.info(f"Best pipeline {self.objective.name}: {best_pipeline['score']:3f}")
+        if self._best_pipeline is not None:
+            best_pipeline = self.rankings.iloc[0]
+            best_pipeline_name = best_pipeline["pipeline_name"]
+            logger.info(f"Best pipeline: {best_pipeline_name}")
+            logger.info(f"Best pipeline {self.objective.name}: {best_pipeline['score']:3f}")
         self._searched = True
 
     def _find_best_pipeline(self):
         """Finds the best pipeline in the rankings
         If self._best_pipeline already exists, check to make sure it is different from the current best pipeline before training and thresholding"""
+        if len(self.rankings) == 0:
+            return
         best_pipeline = self.rankings.iloc[0]
         if not (self._best_pipeline and self._best_pipeline == self.get_pipeline(best_pipeline['id'])):
             self._best_pipeline = self.get_pipeline(best_pipeline['id'])
             if self._train_best_pipeline:
                 X_threshold_tuning = None
                 y_threshold_tuning = None
                 X_train, y_train = self.X_train, self.y_train
-                if self.optimize_thresholds and self.objective.is_defined_for_problem_type(ProblemTypes.BINARY) and self.objective.can_optimize_threshold and is_binary(self.problem_type):
+                if is_binary(self.problem_type) and self.objective.is_defined_for_problem_type(self.problem_type) \
+                   and self.optimize_thresholds and self.objective.can_optimize_threshold:
                     X_train, X_threshold_tuning, y_train, y_threshold_tuning = split_data(X_train, y_train, self.problem_type,
                                                                                           test_size=0.2,
                                                                                           random_seed=self.random_seed)
                 self._best_pipeline.fit(X_train, y_train)
-                self._best_pipeline = self._tune_binary_threshold(self._best_pipeline, X_threshold_tuning, y_threshold_tuning)
+                tune_binary_threshold(self._best_pipeline, self.objective, self.problem_type, X_threshold_tuning, y_threshold_tuning)
 
-    def _tune_binary_threshold(self, pipeline, X_threshold_tuning, y_threshold_tuning):
-        """Tunes the threshold of a binary pipeline to the X and y thresholding data
+    def _num_pipelines(self):
+        """Return the number of pipeline evaluations which have been made
 
-        Arguments:
-            pipeline (Pipeline): Pipeline instance to threshold
-            X_threshold_tuning (ww.DataTable): X data to tune pipeline to
-            y_threshold_tuning (ww.DataColumn): Target data to tune pipeline to
+        Returns:
+            int: the number of pipeline evaluations made in the search
+        """
+        return len(self._results['pipeline_results'])
+
+    def _should_continue(self):
```

Comment:
I kept most of the code from `_check_stopping_condition` here, no intended change in behavior, the diff just looks kinda ugly.